cx/dependency_conflict_predictor.py

```
@@ -0,0 +1,356 @@
+            )
+
+    # 2) Reverse dependency check: existing packages requiring a conflicting range.
+    req_map = {name: constraint for name, constraint in requested}
+    for dist in importlib.metadata.distributions():
+        parent = (dist.metadata.get("Name") or "").lower()
+        if not parent:
+            continue
+        requires = dist.requires or []
+        for req in requires:
+            dep_name, dep_constraint = _parse_req_name_and_constraints(req)
+            if dep_name in req_map and req_map[dep_name]:
```

Ruff (0.15.2)

[warning] Unnecessary key check before dictionary access

Replace with `dict.get`

(RUF019)